### PR TITLE
Issue/8241 > master

### PR DESCRIPTION
**Pull request by davidgrindmill**
Front page theming.

### CODE REVIEW

File: config/sync/views.view.landing_page.yml [hunk 1]
```diff
@@ -169,26 +169,9 @@ display:
           entity_type: node
           entity_field: langcode
       pager:
-        type: full
+        type: none
```
Comment (gkurucz) [inline config/sync/views.view.landing_page.yml]: Hmm, is this view used anywhere at all? As I can see this view was the original front page but if I'm not mistaken you no longer use this and in that case this can be removed.

File: web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss [hunk 2]
```diff
@@ -3,3 +3,9 @@
   margin-bottom: 3rem;
   opacity: 0.5;
 }
+
+.landing-page__body {
+  text-align: center;
+  margin-left: 6rem;
```
Comment (gkurucz) [inline web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss]: Have you validated if this looks properly on mobile devices?

File: web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--block-2.html.twig [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{#
```
Comment (gkurucz) [inline web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--block-2.html.twig]: Let's give this block (`block-2`) and all displays inside the `recipe_list` view a more meaningful machine name. You can change the machine name of a view display at the advanced section when editing a view.

If this block is used somewhere (e.g. on the landing page's view display or the block layout) after changing the machine name it will have to be re-added and re-exported.

**Review by gkurucz — CHANGES_REQUESTED** [review]

File: config/sync/views.view.landing_page.yml (hunk 1 above)
Comment (davidgrindmill) [reply config/sync/views.view.landing_page.yml]: Deleted.

File: web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss (hunk 2 above)
Comment (davidgrindmill) [reply web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss]: I've added a media query to improve the appearance on smaller screens.

File: web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--block-2.html.twig (hunk 3 above)
Comment (davidgrindmill) [reply web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--block-2.html.twig]: Done! Will push updates in a few minutes.

File: web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss [hunk 4]
```diff
@@ -3,3 +3,16 @@
...
+  margin-right: 6rem;
+}
+
+@media (max-width: 599px) {
```
Comment (gkurucz) [inline web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss]: Let's move media queries to the next level. There is a very good article about [how to do breakpoints correctly](https://www.freecodecamp.org/news/the-100-correct-way-to-do-css-breakpoints-88d6a5ba1862/).

Also, as we are using SASS you can utilize one of its best functionalities, mixins. This article proved the exact tools you need to use breakpoints with mixins.

Another thing to note: when styling elements, it's a good practice to write css with a 'mobile first' approach, meaning that the default style (the one without media queries) is the one that works for mobile. Then you can begin using mixins for larger screens (min-width 600px, min-width 900px etc.) to adjust the styling for larger screens.

Comment (davidgrindmill) [reply web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss]: I've added some modifications. However, I chose not to use mixins here, because my teammate Mari will be doing site-wide theming after the rest of us finish theming the individual pages separately.

Comment (gkurucz) [reply web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss]: I suggest you create the mixins anyway, because I'd suggest Mari to use the mixins as well, you can help her work if you add them now. :slightly_smiling_face: 

There's an [existing empty mixins file in your theme](https://github.com/Pronovix/food/blob/master/web/themes/custom/recipe_theme/src/sass/base/_mixins.scss), you can use this file to put the mixins in.

And to make this code part look better, you don't need to redefine the class every time, you can also add the mixins or media queries inside the class. E.g.:

```sass
.landing-page__body {
  default rules...

  @include mixin1 {
    specific rules...
  }

  @media (min-width: 600px) {
    other specific rules...
  }
}
```

Also, you don't need the `max-width: 599px` media query as you already have rules that override this if you are viewing larger screens. It's enough to have the mobile rules like at the `default rules...` section above.

File: web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--recent_recipes.html.twig [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+<div class="row">
+
+{% for row in rows %}
+  {%
+    set row_classes = [
+      'col-sm-12 col-md-4 recipe'
+    ]
+  %}
+  <div{{ row.attributes.addClass(row_classes) }}>
+      {{- row.content -}}
```
Comment (gkurucz) [inline web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--recent_recipes.html.twig]: Indentation = 2 spaces

File: web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--recent_recipes.html.twig [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{#
```
Comment (gkurucz) [inline web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--recent_recipes.html.twig]: This file name looks strange, are you sure it works? There shouldn't be underscores in twig file names.

File: web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss [hunk 7]
```diff
@@ -3,3 +3,31 @@
...
+    margin-left: 5rem;
+    margin-right: 5rem;
+  }
+  @include for-big-desktop-up {
+    margin-left: 6rem;
+    margin-right: 6rem;
+  }
+}
+
+.landing-page img {
```
Comment (gkurucz) [inline web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss]: Please use nesting everywhere for the sake of consistency.

```sass
.outer-selector {
  .inner-selector {
    ...rules
  }
}
```

File: web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss [hunk 8]
```diff
@@ -3,3 +3,31 @@
   margin-bottom: 3rem;
   opacity: 0.5;
 }
+
+.landing-page__body {
+  text-align: center;
+  @include for-phone-only {
```
Comment (gkurucz) [inline web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss]: If these are the default rules, do you need to specify that these are explicitly for very small screens?

**Review by gkurucz — CHANGES_REQUESTED** [review]

File: web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--recent_recipes.html.twig (hunk 5 above)
Comment (davidgrindmill) [reply web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--recent_recipes.html.twig]: Fixed.

File: web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--recent_recipes.html.twig (hunk 6 above)
Comment (davidgrindmill) [reply web/themes/custom/recipe_theme/templates/views/views-view-unformatted--recipe-list--recent_recipes.html.twig]: It did work, but I changed it to a dash and it still works.

File: web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss (hunk 7 above)
Comment (davidgrindmill) [reply web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss]: Fixed.

File: web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss (hunk 8 above)
Comment (davidgrindmill) [reply web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss]: Sorry, I'm not sure what the issue is here. Could you please elaborate?
Edit: nevermind, I updated this section.

**Review by gkurucz — DISMISSED** [review]

File: web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss [hunk 9]
```diff
@@ -1,5 +1,29 @@
-.landing-page__field-image-hero {
-  margin-top: 3rem;
-  margin-bottom: 3rem;
-  opacity: 0.5;
+.landing-page__body {
+  text-align: center;
+  margin-top: 2.5rem;
```
Comment (gkurucz) [inline web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss]: These rules can be simplified as

`margin: 2.5rem 2rem;`

**Review by gkurucz — CHANGES_REQUESTED** [review]

File: web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss (hunk 9 above)
Comment (davidgrindmill) [reply web/themes/custom/recipe_theme/src/sass/content/_node.landing-page.scss]: Fixed

File: config/sync/system.site.yml [hunk 10]
```diff
@@ -5,7 +5,7 @@ slogan: 'Your Dream Dish'
 page:
   403: ''
   404: ''
-  front: /node
```
Comment (dzadza) [inline config/sync/system.site.yml]: please revert this change

File: config/sync/views.view.landing_page.yml [hunk 11]
```diff
@@ -1,311 +0,0 @@
-uuid: 34b1c469-ffe4-4eed-9a86-0e1d995ee4e0
```
Comment (dzadza) [inline config/sync/views.view.landing_page.yml]: I don't think this view should be deleted.

**Review by dzadza — CHANGES_REQUESTED** [review]

**Review by dzadza — APPROVED** [review]

**Review by gkurucz — APPROVED** [review]
